File: lib/Model/review.dart
import 'package:app/Model/talk.dart';
import 'package:cloud_firestore/cloud_firestore.dart';

class Review {
  String idDoc;

  String authorUsername;

  Timestamp createdAt;

  Talk talk;
  String talkid;

  String body;

  Review({this.authorUsername, this.createdAt, this.talkid});

  Review.fromMap(Map<dynamic, dynamic> data) {
    idDoc = data['idDoc'];
    authorUsername = data['Author'];
    body = data['Body'];
    createdAt = data['CreatedAt'];
    talkid = data['Talk'];
  }

  Map<String, dynamic> toMap(){
    return {
      'idDoc' : idDoc,
      'Body' : body,
      'Author' : authorUsername,
      'CreatedAt' : createdAt,
      'Talk' : talkid,
    };
  }

}

File: backend/open_cx_server/models/review.js
const mongoose = require('mongoose');

const reviewSchema = mongoose.Schema({
  idDoc: {
    type: String,
    required: true
  },
  authorUsername: {
    type: String
  },
  createdAt: {
    type: String,
  },
  talkID: {
    type: String,
    required: true,
  },
  body: {
    type: String,
    required: true,
  }
});

module.exports = review = mongoose.model('Review', reviewSchema);
